components/core/src/clp/NetworkReader.cpp

```diff
@@ -130,7 +131,12 @@ NetworkReader::NetworkReader(
     for (size_t i = 0; i < m_buffer_pool_size; ++i) {
         m_buffer_pool.emplace_back(m_buffer_size);
     }
-    m_downloader_thread = std::make_unique<DownloaderThread>(*this, offset, disable_caching);
+    m_downloader_thread = std::make_unique<DownloaderThread>(
+            *this,
+            offset,
+            disable_caching,
+            http_header_kv_pairs
```

Comment:
```suggestion
            std::move(http_header_kv_pairs)
```
Similar to the previous comment